Mathlib/RingTheory/PicardGroup.lean

```diff
@@ -524,7 +518,28 @@ instance (R) [CommRing R] [Finite (MaximalSpectrum R)] : Subsingleton (Pic R) :=
   subsingleton_iff.mpr fun _ _ _ _ ↦ free_of_flat_of_finrank_eq _ _ 1
     fun _ ↦ let _ := @Ideal.Quotient.field; Invertible.finrank_eq_one ..
 
-end CommRing.Pic
+variable (R) (A : Type*) [CommSemiring A]
+
+open AlgebraTensorModule in
+/-- Every `R`-algebra `A` gives rise to a homomorphism between Picard groups of `R` and `A`. -/
+@[simps] noncomputable def mapAlgebra [Algebra R A] : Pic R →* Pic A where
```

Comment:
I find this slightly odd taking in `Algebra R A` but spitting out a map. I would assume it should either take a ring hom or spit out `MulAction (Pic R) (Pic A)`.

Reply:
MulAction is much weaker? I think it's much more common to apply this to an algebra rather than ring hom. I originally defined `mapRingHom` from this using `RingHom.toAlgebra`, but later deleted it as I'm not using it in the foreseeable future. What do people do with the Brauer group?

Reply:
I would like to have a version for ring homs. If you think the algebra version is useful (which I believe) then let's also keep this one.
But having `Pic` as a functor is worthwhile.